46_revocation_severity_and_context.md

```diff
@@ -0,0 +1,301 @@
+   revocation event that is issued by Keylime.
+  * Tag all parts that cause revocations in Keylime with an unique id.
+  * Assign context provided by the user to (static) revocation events.
+  * Provide the option for Keylime to add context itself to revocation events.
+    This is useful if for example Eventlog parsing gains support for dynamic
+    policies.
+  * Extend the verifier API and tenant to add those rules to Keylime.
+ * Provide API entry point for checking which revocations events were triggered
+   for a agent.
+
+### Non-Goals
+ * The classification of revocation events is highly environment and
+   configuration specific. It is not a goal to provide a default configuration
+   other than all events are classified with the highest severity.
+
+
+## Proposal
+Currently revocation events are either send if something with the communication
+the the agent went wrong, the quote is not valid or one of the PCR based checks
+fail.
+
+This proposal changes that behavior and introduces the concept of severity
+levels of events and adds the option to provide more context for events.
+
+All parts that cause revocation events are tagged with an unique id and against
+those ids user scan specify a severity level and provide some context information.
+
+Instead of stop polling an agent if a failure occurred the agent is added back
+for checking if the failure is recoverable. If it is not a revocation event with
+the highest severity is generated.
+
+All the send events are visible in the verifier API for other systems to use.
+
+Keylime also gains the ability to specify context for a revocation event.
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this enhancement is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+The option to classify the revocation events opens the possibility that events
+are not handled. To mitigate this all revocation events that are not explicitly
+from the user classified are assigned the highest severity level.
+
+The same concept applies for revocation events that are caused failures that
+make the agent irrecoverable. Those also always need to be treated with the
+highest severity by default because otherwise some events wont be caught is a
+irrecoverable failure is triggered before.
+
+## Design Details
+We keep the current model of the states, but modify the behavior of the failure
+states. If we are in a failure state that is not irrecoverable the polling of
```

Comment:
Instead of the double negative "not irrecoverable" can we change that to "recoverable"?

Reply:
fixed.